From 45f5082749ab6509c59833ca6269ca4f140f4b3a Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Tue, 26 Dec 2017 12:39:21 +0800
Subject: [PATCH 2134/2241] dts: a8k: mcbin: enable pcie serdes init in kernel

Since the A8K Macchiatobin board has the EP reset
connected to GPIO, which is needed by the PCIe serdes
power up/down sequence from HW guidelines, so it is
necessary to support PCIe serdes initialization in kernel.

To initialize serdes in kernel, it need to power off
serdes first to remove dependency on bootloader, and
then implement the PCIe serdes power up/down sequence
with COMPHY lanes that is binded in DTS.

There is a property of skip_pcie_power_off to skip PCIe
serdes power off in kernel and keep the configuration from
bootloader, so for Macchiatobin board the property should
be removed. For other boards, such as A7/8K-DB, the
property is still retained and depend on bootloader
to initialize PCIe serdes.

The patch remvoed the property of skip_pcie_power_off
for A8K Macchiatobin.

Change-Id: I9614682802c64b3488e582164f12210f09c87b93
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48119
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3900-db.dtsi | 1 +
 arch/arm64/boot/dts/marvell/armada-7020-amc.dts | 1 +
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi | 3 +++
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi | 6 ++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   | 1 -
 5 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi b/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
index 84f8676..e2e0fa5 100644
--- a/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
@@ -181,6 +181,7 @@
 				};
 			};
 			comphy {
+				skip_pcie_power_off;
 				status = "okay";
 			};
 		};
diff --git a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
index 8bc0361..3e28b93 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
@@ -155,6 +155,7 @@
 			};
 
 			comphy {
+				skip_pcie_power_off;
 				status = "okay";
 			};
 		};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 95c15fe..d9bcca3 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -146,6 +146,9 @@
 					reg = <0x1000000 0x3f000000>;
 				};
 			};
+			cp0_comphy: comphy {
+				skip_pcie_power_off;
+			};
 		};
 	};
 	usb3h0_phy: usb3-h0-phy {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 1093dac..682a8af 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -176,6 +176,9 @@
 				non-removable;
 				pinctrl-0 = <&cp0_sdhci_pins>;
 			};
+			cp0_comphy: comphy {
+				skip_pcie_power_off;
+			};
 		};
 
 		cp0_usb3h0_phy: cp0_usb3_phy0 {
@@ -278,6 +281,9 @@
 					};
 				};
 			};
+			cp1_comphy: comphy {
+				skip_pcie_power_off;
+			};
 		};
 
 		cp1_usb3h0_phy: cp1_usb3_phy0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 21f25d7..d05c4f2 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -550,7 +550,6 @@
 				dfx-reg = <0x400280 4>;
 				#phy-cells = <2>;
 				sar-data = <&CP110_LABEL(sar)>;
-				skip_pcie_power_off;
 				status = "disabled";
 			};
 
-- 
2.7.4

